### ISSUE TITLE
KeyError: 'filename' running simple inference example

### ISSUE THREAD

**Issue by scasasso** [open-mmlab/mmdetection#1254]
Hi,
after successful installation, I am trying to go through the inference example in:
https://github.com/open-mmlab/mmdetection/blob/master/docs/GETTING_STARTED.md#high-level-apis-for-testing-images

However, at line:
```python
result = inference_detector(model, img)
```

I get:
```python
KeyError                                  Traceback (most recent call last)
<ipython-input-1-8b1fc4b49116> in <module>
     10 # test a single image and show the results
     11 img = 'test.jpg'  # or img = mmcv.imread(img), which will only load it once
---> 12 result = inference_detector(model, img)

~/Kaggle/open_images_2019_object_detection/mmdetection/mmdet/apis/inference.py in inference_detector(model, img)
     75     # prepare data
     76     data = dict(img=img)
---> 77     data = test_pipeline(data)
     78     data = scatter(collate([data], samples_per_gpu=1), [device])[0]
     79     # forward the model

~/Kaggle/open_images_2019_object_detection/mmdetection/mmdet/datasets/pipelines/compose.py in __call__(self, data)
     22     def __call__(self, data):
     23         for t in self.transforms:
---> 24             data = t(data)
     25             if data is None:
     26                 return None

~/Kaggle/open_images_2019_object_detection/mmdetection/mmdet/datasets/pipelines/test_aug.py in __call__(self, results)
     23                 _results['scale'] = scale
     24                 _results['flip'] = flip
---> 25                 data = self.transforms(_results)
     26                 aug_data.append(data)
     27         # list of dict to dict of list

~/Kaggle/open_images_2019_object_detection/mmdetection/mmdet/datasets/pipelines/compose.py in __call__(self, data)
     22     def __call__(self, data):
     23         for t in self.transforms:
---> 24             data = t(data)
     25             if data is None:
     26                 return None

~/Kaggle/open_images_2019_object_detection/mmdetection/mmdet/datasets/pipelines/formating.py in __call__(self, results)
    147         img_meta = {}
    148         for key in self.meta_keys:
--> 149             img_meta[key] = results[key]
    150         data['img_meta'] = DC(img_meta, cpu_only=True)
    151         for key in self.keys:

KeyError: 'filename'
```

Any idea what may cause this?

Thanks!

**Comment by hellock** [comment]
Thanks for report! It should be fixed in #1255.

**Comment by scasasso** [comment]
yup, it works now!
Thanks!
